### PR TITLE
Refactored sitemap generation functions

### PR DESCRIPTION
**Pull request by toiletspider**
### References Issues:
Sitemap: Remove private events, Filter dates, Add venue, city and states [SEO]

### Description:
Ticket required that venue, city and state pages to be added to site map. These pages does not exist yet, but will be created at a later date. Code was refactored to ease the adding of these to the sitemap.

### Migrations
 * No Migrations

### Environment Variables
 * No change

### CODE REVIEW

File: api/src/controllers/sitemap_gen.rs [hunk 1]
```diff
@@ -2,67 +2,77 @@ use actix_web::{HttpResponse, State};
...
+        "tickets".to_string(),
+        conn,
+    );
 
-    let mut urls = slugs
-        .iter()
-        .map(|slug| format!("{}/tickets/{}", state.config.front_end_url, &slug.slug))
-        .collect_vec();
+    // Cities
+    let city_slug_id = Slug::find_by_slug("city", conn)?.iter().map(|s| s.id).collect_vec();
```
Comment (mikethetike) [inline api/src/controllers/sitemap_gen.rs]: Find_slug_type maybe?

**Comment by Krakaw** [comment]
Are we going to do the sitemap.xml file caching in another PR?

**Comment by toiletspider** [comment]
@Krakaw we are will not be doing caching since it provides no real benefits and only adds complexity. This end point will only be hit a few times a week

File: api/src/controllers/genres.rs [hunk 2]
```diff
@@ -8,3 +8,17 @@ pub fn index(connection: Connection) -> Result<HttpResponse, BigNeonError> {
     Ok(HttpResponse::Ok()
         .json(json!({"genres": &genres.into_iter().map(|g| g.name).collect::<Vec<String>>()})))
 }
+
+#[derive(Serialize)]
+struct GenreListItem {
```
[the remark this answers is not in the document]
Comment (toiletspider) [reply api/src/controllers/genres.rs]: this has been moved to another

**Comment by Krakaw** [comment]
> @Krakaw we are will not be doing caching since it provides no real benefits and only adds complexity. This end point will only be hit a few times a week

From a DDoS perspective this might land up being quite a db intensive operation and is publicly accessible. Though we can add it to our caching strategy with redis when that comes to be.

**Pull request merged**